src/families/cosmos/types.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type { BigNumber } from "bignumber.js";
+import type {
+  TransactionCommon,
+  TransactionCommonRaw
+} from "../../types/transaction";
+
+export type CoreStatics = {};
+export type CoreAccountSpecifics = {};
+export type CoreOperationSpecifics = {};
+export type CoreCurrencySpecifics = {};
+
+export type CosmosDelegation = {|
+  validatorAddress: string,
+  amount: BigNumber,
+  pendingRewards: BigNumber,
+  isActive: boolean, // TBD is there more status case? if so, switch to status field with an enum string.
+|};
+
+export type CosmosRedelegation = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosUnbounding = {|
+  validatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosResources = {|
+  delegations: CosmosDelegation[],
+  redelegations: CosmosRedelegation[],
+  unbounding: CosmosUnbounding[],
+  delegatedBalance: BigNumber,
+  pendingRewardsBalance: BigNumber, 
+  unboundingBalance: BigNumber,
+  withdrawAddress: string,
+|};
+
+export type CosmosDelegationRaw = {|
+  validatorAddress: string,
+  amount: string,
+  pendingRewards: string,
+  isActive: boolean,
+|};
+
+export type CosmosRedelegationRaw = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosUnboundingRaw = {|
+  validatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosResourcesRaw = {|
+  delegations: CosmosDelegationRaw[],
+  redelegations: CosmosRedelegationRaw[],
+  unbounding: CosmosUnboundingRaw[],
+  delegatedBalance: string,
+  pendingRewardsBalance: string, 
+  unboundingBalance: string,
+  withdrawAddress: string,
+|};
+
+// NB this must be serializable (no Date, no BigNumber)
+export type CosmosValidatorItem = {|
+  validatorAddress: string,
+  name: string,
+  votingPower: number, // value from 0.0 to 1.0 (normalized percentage)
+  commission: number, // value from 0.0 to 1.0 (normalized percentage)
+|};
+
+// by convention preload would return a Promise of CosmosPreloadData
+export type CosmosPreloadData = {
+  validators: CosmosValidatorItem[]
+};
+
+export type CosmosOperationMode =
+  | "send"
+  | "delegate"
+  | "undelegate"
+  | "redelegate"
+  | "claimReward"
+  | "claimAllRewards";
```

Comment:
There's no `claimAllRewards`, nor will we support the REST API feature for that ; at least for the moment. I suppose it can stay there because it's a model and we don't want to change it too often, I'm just pointing this out.

If you want to support all types of operations on the network the list will have to be bigger, we can discuss it later (Basically, we'd need one OperationMode per CosmosMsgType here : https://github.com/gagbo/lib-ledger-core/blob/18e81d2626bd26b2a9450ee7fecad805d76edeb7/core/idl/wallet/cosmos/messages.djinni#L126-L141 - without the validator-specific messages I suppose)

Reply:
ok! this was taken from the current wip work done but we can remove that one on both side then.

Reply:
Sorry it was a try to test all the pending rewards, it will be removed !